main.rb

```diff
@@ -0,0 +1,570 @@
+  def my_none?(param = nil)
+    arr = self
+    any_h = { f_b_e: 0, f_e: 0, f_rxp: 0, f_p: 0, who: true }
+
+    0.upto(arr.length - 1) do |i|
+      if param.nil?
+        if block_given?
+          any_h[:f_b_e] += 1 unless yield arr[i] # counts false block's element
+        elsif arr[i].nil? || arr[i] == false
+          any_h[:f_e] += 1 # counts false element
+        end
+      end
+      any_h[:f_rxp], any_h[:f_p] = param_reg(param, arr[i], any_h[:f_rxp], any_h[:f_p], any_h[:who])
+    end
+    any_element_true(arr.length, any_h[:f_b_e], any_h[:f_e], any_h[:f_rxp], any_h[:f_p])
+  end
+
+  # my_map method definition (updated after point 10 of assigment)
+  def my_map(*my_proc)
+    caller = self
+    return to_enum :my_map_proc unless block_given? || my_proc[0].class == Proc
+
+    back = []
+    caller.my_each do |value|
+      if my_proc[0].is_a? Proc
+        back.push my_proc[0].call value
+      elsif block_given?
+        back.push(yield value)
+      end
+    end
+    back
+  end
+
+  # my_count method definition
+  def my_count(param = '')
+    caller = self
+    back = 0
+    caller.my_each do |value|
+      if block_given?
+        back += 1 if yield value
+      elsif param != '' && !block_given?
+        back += 1 if value == param
+      else
+        back += 1
+      end
+    end
+    back
+  end
+
+  def conditions(con1, con2, con3)
+    # puts "hey I am here, con1: #{con1}, con2: #{con2}, con3: #{con3}"
+    con1 && con2 && con3
+  end
+
+  def assign_if_true(total, value, eval1, eval2)
+    if eval1 && eval2
+      value
+    else
+      total
+    end
+  end
+
+  def inject_total_param(total = [], param = [])
+    if !param.nil? && param.my_any?(Symbol)
+      total[0] = assign_if_true(total[0], 0, !param.my_any?(Numeric), param[0] == :+ || param[0] == :-)
+      total[0] = assign_if_true(total[0], 1, !param.my_any?(Numeric), !(param[0] == :+ || param[0] == :-))
+      total[0] = assign_if_true(total[0], param[0], param.my_any?(Numeric), true)
+      param[0] = assign_if_true(param[0], param[1], param.my_any?(Numeric), true)
+    end
+    [total, param]
+  end
+
+  def zero_value_inject(total, iii, param, block, value)
+    if conditions(iii.zero?, !param.nil?, param[0].is_a?(Numeric))
+      total[0] = param[0]
+    elsif conditions(iii.zero?, block, value.is_a?(Numeric))
+      total[0] = value - value
+    elsif conditions(iii.zero?, block, !value.is_a?(Numeric))
+      total[0] = value.clear
+    end
+    total
+  end
+
+  # my_inject method definition
+  def my_inject(*param)
+    caller = self
+    i = 0
+    total = []
+    caller.my_each do |value|
+      total = zero_value_inject(total, i, param, block_given?, value)
+
+      if !param.nil? && param.my_any?(Symbol)
+        total, param = inject_total_param(total, param)
+        total[i + 1] = total[i].send(param[0], value)
+      else
+        total[i + 1] = yield total[i], value
+      end
+
+      i += 1
+    end
+    total[-1]
+  end
```

Comment:
For `my_inject` handle the case when a block is passed with a binary operation, it should combine all elements of enum by applying the binary operation.

- For built-in method
 `%w[dog door rod blade].inject{ |memo, word| memo.length > word.length ? memo : word }` returns `blade`

- Your method gives the following error

> FrozenError:
       can't modify frozen String

Reply:
- Check this [doc](https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-inject) for more information on `inject` method.

Reply:
Hi, I cannot replicate this behaviour. I believe there is a syntax error in the included "built-in method" as it should be "return" and not return. Where is this code located?
If I run the code in the command line (IRB) I get the following results. 

irb(main):222:0> %w[dog door rod blade].my_inject{ |memo, word| memo.length > word.length ? memo : word } 
=> "blade"
irb(main):223:0> %w[dog door rod blade].inject{ |memo, word| memo.length > word.length ? memo : word } 
=> "blade"

And so, I don't see an issue here. Can you be more specific on now my_inject is failing?

Reply:
Hi, So sorry about this.
You are right, the error isn't caused by your method but by the `frozen_string_literal: true` comment.